mantidproject/mantid
# Allow multiple goniometers to be stored in the Run object

**Description of work.**

This is mainly useful for constant wavelength instruments that do motor scans.

**To test:**

Both of these datasets are included in the unit test data.

```python
ws = LoadMD('HB3A_data.nxs')
SetGoniometer(ws, Axis0='omega,0,1,0,-1', Axis1='chi,0,0,1,-1', Axis2='phi,0,1,0,-1', Average=False)
r = ws.getExperimentInfo(0).run()
for i in range(r.getNumGoniometers()):
    print(i,r.getGoniometer(i).getEulerAngles('YZY'))
```

you should get

```
0 [167.591,36.3967,-80.0522]
1 [167.491,36.3967,-80.0522]
2 [167.391,36.3967,-80.0522]
3 [167.291,36.3967,-80.0522]
4 [167.191,36.3967,-80.0522]
5 [167.091,36.3967,-80.0522]
6 [166.991,36.3967,-80.0522]
7 [166.891,36.3967,-80.0522]
8 [166.791,36.3967,-80.0522]
9 [166.691,36.3967,-80.0522]
10 [166.591,36.3967,-80.0522]
11 [166.491,36.3967,-80.0522]
12 [166.391,36.3967,-80.0522]
```

```python
ws = LoadILLDiffraction(Filename='ILL/D20/000017.nxs')
SetGoniometer(ws, Axis0='omega.position,0,1,0,1', Average=False)
for i in range(ws.run().getNumGoniometers()):
    print(f'{i} omega = {ws.run().getGoniometer(i).getEulerAngles("YZY")[0]:.1f}')
```
you should get

```
0 omega = 1.0
1 omega = 1.2
2 omega = 1.4
3 omega = 1.6
4 omega = 1.8
5 omega = 2.0
6 omega = 2.2
7 omega = 2.4
8 omega = 2.6
9 omega = 2.8
10 omega = 3.0
11 omega = 3.2
12 omega = 3.4
13 omega = 3.6
14 omega = 3.8
15 omega = 4.0
16 omega = 4.2
17 omega = 4.4
18 omega = 4.6
19 omega = 4.8
20 omega = 5.0
```

*There is no associated issue.*



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
